File: frontend/models/Report.js
import mongoose from 'mongoose'

const ReportSchema = new mongoose.Schema({
  title: { type: String, required: true },
  type: { type: String, required: true },
  generatedBy: { type: mongoose.Schema.Types.ObjectId, ref: 'User', required: true },
  countries: [{
    name: String,
    isoCode: String
  }],
  dateRange: {
    start: { type: Date, required: true },
    end: { type: Date, required: true }
  },
  summary: { type: String, default: '' },
  data: { type: mongoose.Schema.Types.Mixed, default: {} },
  format: { type: String, default: 'pdf' },
  status: { type: String, default: 'completed' },
  fileUrl: { type: String, default: '' },
  metrics: {
    totalReadings: { type: Number, default: 0 },
    anomaliesDetected: { type: Number, default: 0 },
    avgTemperature: { type: Number, default: 0 },
    avgCO2: { type: Number, default: 0 }
  }
}, {
  timestamps: true,
  collection: 'reports'
})

export default mongoose.models.Report || mongoose.model('Report', ReportSchema)

File: backend/models/report.py
from pydantic import BaseModel, Field
from typing import Optional, Any
from datetime import datetime


class ReportCountry(BaseModel):
    name: str
    isoCode: str


class DateRange(BaseModel):
    start: datetime
    end: datetime


class ReportMetrics(BaseModel):
    totalReadings: int = 0
    anomaliesDetected: int = 0
    avgTemperature: float = 0
    avgCO2: float = 0


class ReportCreate(BaseModel):
    title: str
    type: str  # monthly, quarterly, annual, custom, anomaly, comparison
    generatedBy: str
    countries: list[ReportCountry] = []
    dateRange: DateRange
    summary: str = ""
    data: Any = {}
    format: str = "pdf"
    status: str = "completed"
    fileUrl: str = ""
    metrics: ReportMetrics = ReportMetrics()


class ReportOut(BaseModel):
    id: str = Field(alias="_id")
    title: str
    type: str
    generatedBy: Optional[dict] = None
    countries: list[ReportCountry] = []
    dateRange: DateRange
    summary: str = ""
    data: Any = {}
    format: str = "pdf"
    status: str = "completed"
    fileUrl: str = ""
    metrics: ReportMetrics = ReportMetrics()
    createdAt: Optional[datetime] = None
    updatedAt: Optional[datetime] = None

    class Config:
        populate_by_name = True
